pandas/tests/io/test_parquet.py

```diff
@@ -431,27 +440,39 @@ def test_columns_dtypes_invalid(self, engine):
         self.check_error_on_write(df, engine, ValueError, msg)
 
     @pytest.mark.parametrize("compression", [None, "gzip", "snappy", "brotli"])
-    def test_compression(self, engine, compression):
+    def test_compression(self, engine, compression, request):
 
         if compression == "snappy":
             pytest.importorskip("snappy")
 
         elif compression == "brotli":
             pytest.importorskip("brotli")
 
+        if PY310 and engine == "fastparquet":
+            request.node.add_marker(
+                pytest.mark.xfail(reason="fastparquet failing on 3.10")
+            )
         df = pd.DataFrame({"A": [1, 2, 3]})
         check_round_trip(df, engine, write_kwargs={"compression": compression})
 
-    def test_read_columns(self, engine):
+    def test_read_columns(self, engine, request):
         # GH18154
+        if PY310 and engine == "fastparquet":
```

Comment:
can you just do this in the engine fixture itself?

Reply:
Appears some of the tests parameterize with `engine="fastparquet"` pass and some fail, so these need to be targeted.